File: RestBnb.Domain/Constants/ApiRoutes.cs
namespace RestBnb.Core.Constants
{
    public static class ApiRoutes
    {
        public const string Root = "api";
        public const string Version = "v1";
        public const string Base = Root + "/" + Version;

        public static class Auth
        {
            public const string Login = Base + "/auth/login";
            public const string Register = Base + "/auth/register";
            public const string Refresh = Base + "/auth/refresh";
        }

        public static class Properties
        {
            public const string GetAll = Base + "/properties";
            public const string Create = Base + "/properties";
            public const string Get = Base + "/properties/{propertyId}";
            public const string Update = Base + "/properties/{propertyId}";
            public const string Delete = Base + "/properties/{propertyId}";
        }       
        
        public static class PropertyImages
        {
            public const string GetAll = Base + "/properties/{propertyId}/images";
            public const string Create = Base + "/properties/{propertyId}/images";
            public const string Update = Base + "/properties/{propertyId}/images/{imageId}";
            public const string Delete = Base + "/properties/{propertyId}/images/{imageId}";
        }

        public static class Bookings
        {
            public const string GetAll = Base + "/bookings";
            public const string Create = Base + "/bookings";
            public const string Get = Base + "/bookings/{bookingId}";
            public const string Update = Base + "/bookings/{bookingId}";
            public const string Delete = Base + "/bookings/{bookingId}";
        }

        public static class BookingsList
        {
            public const string GetMyBookings = Base + "/bookingsList";
        }

        public static class Users
        {
            public const string Get = Base + "/users/{userId}";
            public const string Update = Base + "/users/{userId}";
            public const string Delete = Base + "/users/{userId}";
        }

        public static class Cities
        {
            public const string GetAll = Base + "/cities";
            public const string Get = Base + "/cities/{cityId}";
        }

        public static class HealthChecks
        {
            public const string Details = "/health";
        }
    }
}

File: RestBnb/ClientApp/src/app/shared/constants/apiRoutes.ts
import { environment } from '../../../environments/environment';

export class ApiRoutes {
  private static ApiBase = environment.apiUrl;

  static Auth = class {
    public static Login = ApiRoutes.ApiBase + '/auth/login';
    public static Register = ApiRoutes.ApiBase + '/auth/register';
  };

  static Cities = class {
    public static GetAll = ApiRoutes.ApiBase + '/cities';
    public static Get = ApiRoutes.ApiBase + '/cities';
  };

  static Properties = class {
    public static GetAll = ApiRoutes.ApiBase + '/properties';
    public static Create = ApiRoutes.ApiBase + '/properties';
    public static Get = ApiRoutes.ApiBase + '/properties/';
    public static Delete = ApiRoutes.ApiBase + '/properties/';
  };

  static Bookings = class {
    public static GetAll = ApiRoutes.ApiBase + '/bookings';
    public static Create = ApiRoutes.ApiBase + '/bookings';
    public static Put = ApiRoutes.ApiBase + '/bookings/bookingId';
    public static Get = ApiRoutes.ApiBase + '/bookings/';
  };

  static BookingsList = class {
    public static GetMyBookings = ApiRoutes.ApiBase + '/myBookingsList';
    public static GetMyPropertiesBookings =
      ApiRoutes.ApiBase + '/myPropertiesBookingsList';
  };

  static PropertyImages = class {
    public static Create = ApiRoutes.ApiBase + '/properties/propertyId/images';
    public static GetAll = ApiRoutes.ApiBase + '/properties/propertyId/images';
  };
}
